yt_dlp/extractor/youtube.py

```diff
@@ -2091,8 +2092,17 @@ def _extract_comment(self, comment_renderer, parent=None):
             'parent': parent or 'root'
         }
 
-    def _comment_entries(self, root_continuation_data, ytcfg, video_id, parent=None, comment_counts=None):
+    @dataclasses.dataclass
+    class _CommentTracker:
+        running_total: int = 0
+        est_total: int = 0
+        current_page_thread: int = 0
+        total_parent_comments: int = 0
+        total_reply_comments: int = 0
```

Comment:
Anyone know of a better way of going about this? Before we were using a list and it wasn't easily readable. The recursive design of the comment extractor makes keeping track of this data a bit messy.

Reply:
darn, can't use this anyways as dataclasses are not supported in py3.6

Reply:
only way I can think of is to put the entire comment extraction code in a class/closure

Reply:
Yeah :/
maybe that can be a future improvement